src/client/containers/chart-detail-page/chart-detail-page.component.js

```diff
@@ -1,32 +1,55 @@
-import React from 'react';
-import './page-detail-chart.style.css';
-import PropTypes from 'prop-types';
+import React, { useContext } from 'react';
+import '../../components/detail-chart/detail-chart.style.css';
 import DetailChart from '../../components/detail-chart/detail-chart.component';
 import SidebarMenu from '../../components/side-navigation/sidebar.component';
 import ProgressBar from '../../components/progress-bar/progress-bar.component';
-import Footer from '../../components/footer/footer.component';
-import DateRange from '../../components/date-range/date-range.component';
-import data from '../../assets/lettuce.json';
+import './chart-detail-page.css';
+// import UpdateDateRange from '../../components/update-date-range/update-date-range.component';
+import ChartbarMenu from '../../components/chart-data-buttons/chartbar-buttons/chart-data-button.component';
+import { ChartDataContext } from './chart-detail-page.context';
+
+const value = [
+  { id: 1, label: 'Last 5 Days' },
+  { id: 2, label: 'Last Week' },
+  { id: 3, label: 'Last Two Weeks' },
+  { id: 4, label: 'Last Month' },
+  { id: 5, label: 'Custom' },
+];
+
+const ChartDetailPage = () => {
+  const chartvalues = useContext(ChartDataContext);
+  const {
+    boundaryData,
+    materialName,
+    sensorData,
+    startDate,
+    currentDate,
+    stages,
+    units,
+  } = chartvalues;
+
+  const headingText = materialName.toUpperCase();
 
-const ChartDetailPage = ({ title }) => {
   return (
     <>
-      <div>
-        <SidebarMenu isActive={true} isUser={true} />
-      </div>
-      <div>
-        <h1>{title}</h1>
-        <ProgressBar />
-        <DateRange />
-        <DetailChart data={data} />;
-        <Footer />
+      <SidebarMenu />
+      <div className="chart-details">
+        <h1>{headingText} GRAPH DETAILS</h1>
+        <ProgressBar
+          startDate={startDate}
+          currentDate={currentDate}
+          stages={stages}
+        />
+        <ChartbarMenu buttons={value} />
+        <DetailChart
+          data={sensorData}
+          boundary={boundaryData}
+          description={materialName}
+          unit={units}
+        />
       </div>
     </>
   );
 };
 
 export default ChartDetailPage;
-
-ChartDetailPage.propTypes = {
-  title: PropTypes.string.isRequired,
-};
```

Comment:
Please don't remove proptypes. If you got proptype errors it's because you were trying to instantiate components without the props that they need in order to function right.

Reply:
Actually, the prop is coming now from the container component through context API.